### ISSUE TITLE
Make it so that consumers get notifications when filtered search is applied and can access search results

### ISSUE THREAD

**Issue by RyanAtViceSoftware** [GriddleGriddle/Griddle#246]
Given that I've correctly configured Griddle for search notifications
When a user types into the Griddle search box
Then I'm notified that a search was applied
   And I'm able to access the selected rows

**Comment by RyanAtViceSoftware** [comment]
Note that I've created a fork and I'm working on implementing this to support this API contract:

// In MyComponent.render
<Griddle results={myRows} onSearchFilterApplied={this._onSearchApplied}/>

// In MyComponent._onSearchApplied
_onSearchApplied(filteredSearchResults) {
   // filteredSearchResults is an array containing the filtered results rows
}

**Comment by RyanAtViceSoftware** [comment]
I'm getting this error when running 

``` npm install ```

Any ideas?

```
c:\code\Griddle>npm install
npm WARN package.json 6to5-jest@3.0.0 No repository field.
|


> contextify@0.1.14 install c:\code\Griddle\node_modules\jest-cli\node_modules\jsdom\node_modules\contextify
> node-gyp rebuild

/
c:\code\Griddle\node_modules\jest-cli\node_modules\jsdom\node_modules\contextify>node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_mod
ules\node-gyp\bin\node-gyp.js" rebuild
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.Cpp.Platform.targets(64,5): error MSB8020: The build tools for Visual Studio 2010 (Platform To
olset = 'v100') cannot be found. To build using the v100 build tools, please install Visual Studio 2010 build tools.  Alternatively, you may upgrade to the cur
rent Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Upgrade Solution...". [c:\code\Griddle\node_modules\jes
t-cli\node_modules\jsdom\node_modules\contextify\build\contextify.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\12.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "node" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd c:\code\Griddle\node_modules\jest-cli\node_modules\jsdom\node_modules\contextify
gyp ERR! node -v v0.10.36
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok
npm WARN engine hoek@2.16.3: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.36","npm":"1.4.28"})
npm WARN engine boom@2.9.0: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.36","npm":"1.4.28"})
npm WARN engine cryptiles@2.0.5: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.36","npm":"1.4.28"})
npm ERR! contextify@0.1.14 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the contextify@0.1.14 install script.
npm ERR! This is most likely a problem with the contextify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls contextify
npm ERR! There is likely additional logging output above.

npm ERR! System Windows_NT 6.1.7601
npm ERR! command "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! cwd c:\code\Griddle
npm ERR! node -v v0.10.36
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0

c:\code\Griddle>
```

**Comment by ryanlanciaux** [comment]
Does it still run on if you type `grunt`  into the command line? I _think_ that may just be jest failing (you may need to add VS 2012 compiler to path assuming you are on Windows)

**Comment by ryanlanciaux** [comment]
I realize that it has been some time since this issue was active, however, I wanted to respond here and note that we have recently released 1.0 of Griddle. It is possible to interact with the entire state tree of Griddle at any component (through customization) and I imagine the `npm-install` issues posted above should be a less-common thing.

Please [see here for information on customization in Griddle](http://griddlegriddle.github.io/Griddle/docs/customization/) (which references attaching a component to Griddle state).
